FunctionalTableData/CellStyle.swift

```diff
@@ -39,15 +39,16 @@ public struct CellStyle {
 	public var selectionColor: UIColor? = CellStyle.selectionColor
 	/// The view's background color.
 	public var backgroundColor: UIColor?
-	/// The view that is displayed behind the cell’s other content.
-	public var backgroundView: UIView?
+	/// Provides the view that is displayed behind the cell's other content.
+	public var backgroundViewProvider: BackgroundViewProvider
 	/// The tint color to apply to the cell.
 	public var tintColor: UIColor?
 	/// The default spacing to use when laying out content in the view.
 	public var layoutMargins: UIEdgeInsets?
 	/// The radius to use when drawing rounded corners in the view.
 	public var cornerRadius: CGFloat
 	
+	@available(*, unavailable, message: "The `backgroundView` argument is no longer available. Use backgroundViewProvider instead.")
```

Comment:
should mark as deprecated and not unavailable

Reply:
When I mark this deprecated over unavailable I get `ambiguous use of init` wherever we init as `CellStyle()`. Any idea how this can be avoided?